File: leetcode/1441/solution.go
/**
 * https://leetcode.com/problems/build-an-array-with-stack-operations/
 * 1441. Build an Array With Stack Operations
 */

package main

func buildArray(target []int, n int) []string {
	i := 0
	ops := make([]string, 0, len(target))

	for c := 1; c <= n; c++ {
		ops = append(ops, "Push")
		if c != target[i] {
			ops = append(ops, "Pop")
		} else {
			i++
			if i == len(target) {
				break
			}
		}
	}

	return ops
}

File: leetcode/1441/solution.rs
/**
 * https://leetcode.com/problems/build-an-array-with-stack-operations/
 * 1441. Build an Array With Stack Operations
 */

impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut i = 0;
        let mut ops = vec![];

        for current in 1..=n {
            ops.push("Push".to_string());
            if current != target[i] {
                ops.push("Pop".to_string());
            } else {
                i += 1;
                if i == target.len() {
                    break;
                }
            }
        }

        ops
    }
}
